src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -235,13 +104,15 @@ public static int Compare(string? strA, string? strB, StringComparison compariso
                         return strA._firstChar - strB._firstChar;
                     }
 
-                    return CompareOrdinalHelper(strA, strB);
+                    if (strA.Length > 1 && strB.Length > 1)
+                    {
+                        return SpanHelpers.SequenceCompareTo(ref strA.GetRawStringData(), strA.Length, ref strB.GetRawStringData(), strB.Length);
+                    }
 
-                case StringComparison.OrdinalIgnoreCase:
+                    return strA.Length - strB.Length;
```

Comment:
A comment here clarifying that we've reacher here if either `strA` or `strB` is empty, thus we're interested in the length difference only... if `strA` is non-empty, that implied it is greater than the empty `strB` (and vice-versa).

What feels odd to me, in this case, is what was in `strA._firstChar` and `strB._firstChar` in the case of an empty string... because they are being blindly compared up at [line 102](https://github.com/dotnet/runtime/pull/402/files#diff-a70aebfa78baa09ba9bb366e971a14caR102) which seems entirely unsafe.

Reply:
`_firstChar` will be `'\0'` for an empty string as all .NET strings as well as being length prefixed are also null terminated

Reply:
Excellent point. Thus the ONLY way we can reach this line is if `strA` is one or more characters, starting with a leading `'\0'` and `strB` is 0-length (or vice-versa). Do you think this would be clear?
```
// either strA or strB starts with a '\0` character and the other is empty
// return 1 if strA is non-empty or -1 if strB is non-empty
return strA.Length > 0 ? 1 : -1;
```

Reply:
Wouldn't want to change what the return value is; so just -1, 1 as someone *may* have taken a dependency on its semantic meaning.

Have improved comments.